src/download/__init__.py

```diff
@@ -676,19 +690,27 @@ def execute(self, context):
                 }
                 utils.delete_hierarchy(ob)
                 start_download(asset_data, **kwargs)
-        else:
-            kwargs = {
-                'cast_parent': self.cast_parent,
-                'target_object': self.target_object,
-                'material_target_slot': self.material_target_slot,
-                'model_location': tuple(self.model_location),
-                'model_rotation': tuple(self.model_rotation),
-                'replace': False,
-            }
+            return {'FINISHED'}
+
+        target_object = self._get_target_object(context)
+        logging.debug(f'Using target object: {target_object}')
+        kwargs = {
+            'cast_parent': self.cast_parent,
+            'target_object': target_object,
+            'material_target_slot': self.material_target_slot,
+            'model_location': tuple(self.model_location),
+            'model_rotation': tuple(self.model_rotation),
+            'replace': False,
+        }
+        self.target_object = ''
 
-            start_download(asset_data, **kwargs)
+        start_download(asset_data, **kwargs)
         return {'FINISHED'}
 
+    def _get_target_object(self, context):
+        if not self.target_object and context.selected_objects:
```

Comment:
Acho não entendi muito bem o uso do `self.target_object`. Em que casos, ele é relevante?

Reply:
Esse `target_object` é setado no `src/ui/operators/asset_bar` quando o usuário arrasta o material para a cena. Se tiver arrastado para um objeto, esse é o `target_object` e o addon tenta aplicar o material nele
Nesse caso, se o usuário usar o botão "Download to scene",  esse atributo fica vazio